### PR TITLE
Pod Device-Read-BPS support

### PR DESCRIPTION
**Pull request by cdoern**
added the option for the user to specify a rate, in bytes, at which they would like to be able to read from the device being added to the pod. This is the first in a line of pod device options.

WARNING: changed pod name json tag to pod_name to avoid confusion when marshaling with the containerspec's name cc/ @jwhonce 

Signed-off-by: cdoern <cdoern@redhat.com>

### CODE REVIEW

File: docs/source/markdown/podman-pod-create.1.md [hunk 1]
```diff
@@ -41,7 +41,15 @@ Examples of the List Format:
 #### **--device**=_host-device_[**:**_container-device_][**:**_permissions_]
 
 Add a host device to the pod. Optional *permissions* parameter
+<<<<<<< HEAD
```
Comment (rhatdan) [inline docs/source/markdown/podman-pod-create.1.md:44]: Looks like you have a merge error.

Comment (cdoern) [reply docs/source/markdown/podman-pod-create.1.md]: Thanks, I was confused why validation was failing

**Comment by cdoern** [comment]
reliazed the recognition of a device when given through the `--device-read-bps` is different than just by `--device`, need to rework a bit

File: libpod/container_inspect.go [hunk 2]
```diff
@@ -894,3 +870,27 @@ func (c *Container) GetDevices(priv bool, ctrSpec spec.Spec, deviceNodes map[str
 	}
 	return devices, nil
 }
+
+func BlkioDeviceThrottle(deviceNodes map[string]string, devs []spec.LinuxThrottleDevice) ([]define.InspectBlkioThrottleDevice, error) {
```
Comment (mheon) [inline libpod/container_inspect.go:874]: Does this need to be public? If so, can you move it to a utility package (pkg/utils probably best)?

Comment (cdoern) [reply libpod/container_inspect.go]: made it private, it is only used w/in libpod, thanks for catching that.

**Comment by cdoern** [comment]
@mheon @rhatdan  PTAL

**Comment by rhatdan** [comment]
/approve
LGTM
@containers/podman-maintainers PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM, but would like an eyeball from @mheon

File: pkg/specgen/generate/container.go [hunk 3]
```diff
@@ -191,9 +191,6 @@ func CompleteSpec(ctx context.Context, r *libpod.Runtime, s *specgen.SpecGenerat
 	if len(s.User) == 0 && inspectData != nil {
 		s.User = inspectData.Config.User
 	}
-	if err := finishThrottleDevices(s); err != nil {
```
Comment (mheon) [inline pkg/specgen/generate/container.go:194]: Why was this removed?

Comment (mheon) [reply pkg/specgen/generate/container.go:194]: Rather, *moved* - I see it in `container_create`.

Comment (mheon) [reply pkg/specgen/generate/container.go:194]: I think this might break things - when it was here, it ran on the host system, whereas now it runs on the server system (in APIv2 podman-remote scenarios).

Comment (mheon) [reply pkg/specgen/generate/container.go:194]: Then again, that might be a desirable change for OS X...

File: pkg/specgen/podspecgen.go [hunk 4]
```diff
@@ -12,7 +12,7 @@ type PodBasicConfig struct {
 	// Name is the name of the pod.
 	// If not provided, a name will be generated when the pod is created.
 	// Optional.
-	Name string `json:"name,omitempty"`
+	Name string `json:"pod_name,omitempty"`
```
[the remark this answers is not in the document]
Comment (cdoern) [reply pkg/specgen/podspecgen.go:15]: Yes, I can change it to that instead.

File: pkg/specgen/generate/container.go (hunk 3 above)
Comment (cdoern) [reply pkg/specgen/generate/container.go:194]: I had to move is slightly down the call stack because when adding a container to a pod, the specgen doesn't get the pod's devices until `container_create` so `ResourceLimits.BlockIO.ThrottleReadBpsDevice` is filled too early
